xunit-test/CountersTest/StrongCounterTest.cpp

```diff
@@ -0,0 +1,225 @@
+    StrongCounterTest::remoteCacheManager->stop();
+}
+
+TEST_F(StrongCounterTest, testReset) {
+    const std::string counterName("strongTestReset");
+    const long initialValue = 5;
+    auto& counterManager = StrongCounterTest::remoteCacheManager->getCounterManager();
+    auto cc = CounterConfiguration(initialValue, 0, 200, 8, CounterType::BOUNDED_STRONG, Storage::VOLATILE);
+    counterManager.defineCounter(counterName, cc);
+    auto counter = counterManager.getStrongCounter(counterName);
+    EXPECT_EQ(counter->getValue(), 5);
+    EXPECT_EQ(counter->addAndGet(100), 105);
+    counter->reset();
+    EXPECT_EQ(counter->getValue(), initialValue);
+    counter->remove();
+}
+
+TEST_F(StrongCounterTest, testConfiguration) {
+    const std::string counterName("strongTestConfiguration");
+    auto& counterManager = StrongCounterTest::remoteCacheManager->getCounterManager();
+    auto cc = CounterConfiguration(0, -256, 65536, 64, CounterType::BOUNDED_STRONG, Storage::VOLATILE);
+    counterManager.defineCounter(counterName, cc);
+    auto counter = counterManager.getStrongCounter(counterName);
+    CounterConfiguration retCc = counter->getConfiguration();
+    EXPECT_EQ(cc.getLowerBound(), retCc.getLowerBound());
+    EXPECT_EQ(cc.getUpperBound(), retCc.getUpperBound());
+    EXPECT_EQ(0, retCc.getConcurrencyLevel());
+    EXPECT_EQ(cc.getType(), retCc.getType());
+    EXPECT_EQ(cc.getStorage(), retCc.getStorage());
+    EXPECT_EQ(counter->getName(), counterName);
+    counterManager.remove(counterName);
+    counter->remove();
+
+    const std::string counterName1("strongTestConfiguration1");
+    CounterConfiguration cc1 = CounterConfiguration(0, -100, 100, 1, CounterType::UNBOUNDED_STRONG,
+            Storage::PERSISTENT);
+    counterManager.defineCounter(counterName1, cc1);
+    counter = counterManager.getStrongCounter(counterName1);
+    CounterConfiguration retCc1 = counter->getConfiguration();
+    EXPECT_EQ(0, retCc1.getLowerBound());
+    EXPECT_EQ(0, retCc1.getUpperBound());
+    EXPECT_EQ(0, retCc1.getConcurrencyLevel());
+    EXPECT_EQ(cc1.getType(), retCc1.getType());
+    EXPECT_EQ(cc1.getStorage(), retCc1.getStorage());
+    EXPECT_EQ(counter->getName(), counterName1);
+    counterManager.remove(counterName1);
+    counter->remove();
+}
+
+TEST_F(StrongCounterTest, testRemove) {
+    const std::string counterName("strongTestRemove");
+    const long initialValue = 5;
+    auto& counterManager = StrongCounterTest::remoteCacheManager->getCounterManager();
+    auto cc = CounterConfiguration(initialValue, -100, 200, 8, CounterType::UNBOUNDED_STRONG, Storage::VOLATILE);
+    counterManager.defineCounter(counterName, cc);
+    auto counter = counterManager.getStrongCounter(counterName);
+    EXPECT_EQ(counter->addAndGet(100), 105);
+    counter->remove();
+    EXPECT_EQ(counter->getValue(), 5);
+    EXPECT_EQ(counter->addAndGet(-100), -95);
+    counterManager.remove(counterName);
+    EXPECT_EQ(counter->getValue(), 5);
```

Comment:
Shouldn't this usage of the counter after calling `CounterManager.remove(String)` cause an error?

Reply:
No that's ok: the counter is recreated if accessed again. Here's the doc:

http://infinispan.org/docs/dev/user_guide/user_guide.html#hot_rod_protocol_2_7

Reply:
OK, I see now. I think you mean this link:

[http://infinispan.org/docs/dev/user_guide/user_guide.html#remove_a_counter_via_countermanager](http://infinispan.org/docs/dev/user_guide/user_guide.html#remove_a_counter_via_countermanager)

Reply:
Yeah, but this is for the embedded usage, the one above is the for romote via hotrod (see Counter remove operation parag).